tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -334,8 +335,11 @@ def handler(event: Dict[str, Any], context: LambdaContext) -> None:
         raise ValueError(f"Must be passed a single record. Was {len(records)}")
     record = records[0]
     LOGGER.debug(f"Event passed = {event}")
+
+    # pull the body out & json load it
+    record_json = json.loads(record["body"])
     # grab the key_path and bucket name from record
-    key_path = record["s3"]["object"]["key"]
-    bucket_name = record["s3"]["bucket"]["name"]
+    key_path = record_json["Records"][0]["s3"]["object"]["key"]
```

Comment:
Are we confident that there will only ever be one element in "Records"?

Reply:
the sqs has batch size of 1. So maybe yes.

Reply:
There are two arrays in question.
`event["Records"]` which I believe corresponds to entries from SQS (and is thus covered by your batch-size)
and
`json.loads(event["Records"][0]["body"])["Records"]` which I believe corresponds to the message as provided by S3.

Reply:
s3 is sending event notification each time one file is recovered in the bucket. So there will be only 1 element in the event.

Reply:
I mean that's what aws is saying it does

Reply:
In that case, add a link to whatever documentation you've found as a comment in the code, and enforce that 1 record limit with an if-check.

Reply:
added

Reply:
I don't see anything on the linked page indicating that there will always only be one element in "Records".

Reply:
Updated to use looping to capture all records if present